Library/Homebrew/cmd/desc.rb

```diff
@@ -0,0 +1,59 @@
+require 'descriptions'
+require 'cmd/search'
+
+USAGE =<<EOM
+Print formula descriptions.
+
+Usage:
+
+    brew desc --decache             Delete the cache, if present
+    brew desc --recache             [Re]generate the cache
+    brew desc name [name ...]       Describe the named formula(e)
+    brew desc (-s|-n|-d) <string>   Search specifed fields for string
+    brew desc (-s|-n|-d) /<regex>/  Search specifed fields for regex
+EOM
+
+
+module Homebrew
+  def desc
+    options = ARGV.options_only
+
+    if options.include?("--decache")
+      Descriptions.delete_cache
+      options.delete("--decache")
+      cache_op = true
+    end
+
+    if options.include?("--recache")
+      Descriptions.generate_cache
+      options.delete("--recache")
+      cache_op = true
+    end
+
+    if options.empty?
+      if ARGV.named.empty?
+        print USAGE unless cache_op
```

Comment:
We delete the `--delete-cache` (née `--decache`) and `--recache` options after acting upon them, and set `cache_op`. That way we can tell how we came to `options.empty?` case.

Was it because the user called `brew desc` without arguments? Then we whine to some degree. (Although less that we currently do — I’ll be taking @xu-cheng’s advice regarding the usage message.)

Was it because the user called `brew desc` with only cache-administration options, which we’ve already dealt with? Then we don’t whine, and just exit quietly. I hope that makes sense.